src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.cs

```diff
@@ -229,19 +237,26 @@ private void GroupInsertedGraph(List<AnnotationViewModel> existingGroups, string
             DoEvents();
 
             var annotation = this.DynamoViewModel.Model.CurrentWorkspace.AddAnnotation(Resources.InsertedGroupSubTitle, Guid.NewGuid());
-            annotation.AnnotationText = graphName;
-            DoEvents();
+            if (annotation != null)
+            {
+                annotation.AnnotationText = graphName;
+                DoEvents();
```

Comment:
what is up with this use of DoEvents from application code?

Reply:
I have sprinkled them like Magic dust :D I cleaned up the code and got rid of all but one DoEvents(), which I still need in order to get the desired result.